controllers/component_controller.go

```diff
@@ -99,36 +101,68 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	}
 
 	// If the devfile hasn't been populated, the CR was just created
+	var gitToken string
 	if component.Status.Devfile == "" {
+		// If a Git secret was passed in, retrieve it for use in our Git operations
+		// The secret needs to be in the same namespace as the Component
+		if component.Spec.Source.GitSource.Secret != "" {
+			gitSecret := corev1.Secret{}
+			namespacedName := types.NamespacedName{
+				Name:      component.Spec.Source.GitSource.Secret,
+				Namespace: component.Namespace,
+			}
+
+			err = r.Client.Get(ctx, namespacedName, &gitSecret)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Unable to retrieve Git secret %v, exiting reconcile loop %v", component.Spec.Source.GitSource.Secret, req.NamespacedName))
+				r.SetCreateConditionAndUpdateCR(ctx, &component, err)
+				return ctrl.Result{}, nil
+			}
+
+			gitToken = string(gitSecret.Data["password"])
+		}
 		source := component.Spec.Source
 		context := component.Spec.Context
 
 		if source.GitSource != nil && source.GitSource.URL != "" {
 			var devfileBytes []byte
-
+			var gitURL string
 			if source.GitSource.DevfileURL == "" {
-				rawURL, err := util.ConvertGitHubURL(source.GitSource.URL)
-				if err != nil {
-					log.Error(err, fmt.Sprintf("Unable to convert Github URL to raw format, exiting reconcile loop %v", req.NamespacedName))
-					r.SetCreateConditionAndUpdateCR(ctx, &component, err)
-					return ctrl.Result{}, nil
-				}
 
 				// append context to the path if present
 				// context is usually set when the git repo is a multi-component repo (example - contains both frontend & backend)
```

Comment:
can you move this comment to the right place, I was confused for a moment lol

Reply:
Haha sure

Reply:
Updated.